sqlmesh/core/engine_adapter/spark.py

```diff
@@ -443,8 +459,10 @@ def _create_table(
             if isinstance(table_name_or_schema, exp.Schema)
             else exp.to_table(table_name_or_schema)
         )
-        if (kwargs.get("storage_format") or "").lower() == "iceberg" or self.wap_supported(
-            table_name
+        if (
+            not already_exists
+            and (kwargs.get("storage_format") or "").lower() == "iceberg"
+            or self.wap_supported(table_name)
```

Comment:
This condition doesn't look right to me. Wouldn't it always be true if `wap_supported` is true regardless of the `already_exists` value?

Reply:
Thanks should have caught this